pages/api/sessionClient.ts

```diff
@@ -0,0 +1,39 @@
+﻿import {assertTokenIsValid} from "../../helpers/tokenHelpers";
+﻿import getConfig from 'next/config';
+import fetch from "node-fetch";
+import {GetServerSideProps} from "next";
+import {CandidateTestStatus} from "./candidateApiClientModule";
+const {publicRuntimeConfig} = getConfig();
+const baseUrl = publicRuntimeConfig.API_URL;
+export async function checkToken(token: string): Promise<boolean> {
+    const response = await fetch(`${baseUrl}/sessions/${token}`);
+    return response.ok;}
+
+export const getServerSideProps: GetServerSideProps = async ({res, query}) => {
+    await assertTokenIsValid(query, res);
+    return { props: {}};
+};
+export interface SessionCandidate {
+    firstName: string;
+    lastName: string;
+    testStatuses: CandidateTestStatus[];
+}
+export interface NewTestSubmission{
+    testId: number;
+    testAnswer: string;
+}
+
+export async function getSessionCandidate(token: string | string[] | undefined): Promise<SessionCandidate> {
```

Comment:
we always want the token to be a string, so our types should probably reflect that.

Reply:
This was initially to make typescript happy, another formula was using token and expected results of an inbuilt type with the above options so complained that token doesn't have expected characteristics....

Reply:
I will solve other things first then look through and see if there is a way round this as at least it works!